### ISSUE TITLE
Zustand Form Issues

### ISSUE THREAD

**Issue by duskfallcrew** [Ktiseos-Nyx/Ktiseos-Nyx-Trainer#117]
Zustand is NOT saving things properly and our tomls are saving if we hit save config-- but per tab things are not saved in "RUNTIME" like they were promised - although memory serves we were struggling with that so we stopped.

Rather Zustand is only somehow forgetting key things like the dataset folder, because ti's a dropdown/combobox. 

Here's the last known config/toml we tried to make:
 
pretrained_model_name_or_path = "/workspace/Ktiseos-Nyx-Trainer/pretrained_model/Illustrious-XL-v0.1.safetensors"
max_train_epochs = 10
output_dir = "/workspace/Ktiseos-Nyx-Trainer/output"
output_name = "aer0nxvoor_test"
seed = 42
unet_lr = 6e-5
text_encoder_lr = 6e-6
lr_scheduler = "constant"
lr_scheduler_num_cycles = 3
lr_warmup_ratio = 0.1
lr_warmup_steps = 0
lr_power = 1.0
optimizer_type = "CAME"
max_grad_norm = 1.0
weight_decay = 0.1
max_token_length = 225
clip_skip = 2
weighted_captions = false
no_token_padding = false
save_model_as = "safetensors"
save_precision = "fp16"
no_metadata = false
save_every_n_epochs = 1
save_every_n_steps = 0
save_last_n_epochs = 0
save_last_n_epochs_state = 0
save_last_n_steps_state = 0
save_state = false
sample_every_n_epochs = 0
sample_every_n_steps = 0
sample_sampler = "euler_a"
mixed_precision = "fp16"
gradient_checkpointing = true
gradient_accumulation_steps = 1
cache_latents = true
cache_latents_to_disk = false
cache_text_encoder_outputs = false
vae_batch_size = 1
no_half_vae = false
persistent_data_loader_workers = 1
fp8_base = false
full_fp16 = false
lowram = false
xformers = false
sdpa = true
v2 = false
v_parameterization = false
network_train_unet_only = true
noise_offset = 0.0
zero_terminal_snr = false
prior_loss_weight = 1.0
max_data_loader_n_workers = 8
bucket_reso_steps = 64
min_timestep = 0
max_timestep = 1000
v_pred_like_loss = 0.0
vae = "/workspace/Ktiseos-Nyx-Trainer/vae/sdxl_vae.safetensors"
network_module = "networks.lora"
network_dim = 16
network_alpha = 8
conv_dim = 32
conv_alpha = 32

That's the one that was bEFORE the last restart.

And upon refresh we lost all the data including lora name and everthing:

pretrained_model_name_or_path = "/workspace/Ktiseos-Nyx-Trainer/pretrained_model/Illustrious-XL-v0.1.safetensors"
max_train_epochs = 10
output_dir = "/workspace/Ktiseos-Nyx-Trainer"
output_name = "lora"
seed = 42
unet_lr = 0.0001
text_encoder_lr = 5e-5
lr_scheduler = "cosine_with_restarts"
lr_scheduler_num_cycles = 3
lr_warmup_ratio = 0.1
lr_warmup_steps = 0
lr_power = 1.0
optimizer_type = "AdamW8bit"
max_grad_norm = 1.0
weight_decay = 0.1
max_token_length = 225
clip_skip = 2
weighted_captions = false
no_token_padding = false
save_model_as = "safetensors"
save_precision = "fp16"
no_metadata = false
save_every_n_epochs = 1
save_every_n_steps = 0
save_last_n_epochs = 0
save_last_n_epochs_state = 0
save_last_n_steps_state = 0
save_state = false
sample_every_n_epochs = 0
sample_every_n_steps = 0
sample_sampler = "euler_a"
mixed_precision = "fp16"
gradient_checkpointing = true
gradient_accumulation_steps = 1
cache_latents = true
cache_latents_to_disk = false
cache_text_encoder_outputs = false
vae_batch_size = 1
no_half_vae = false
persistent_data_loader_workers = 1
fp8_base = false
full_fp16 = false
lowram = false
xformers = false
sdpa = true
v2 = false
v_parameterization = false
network_train_unet_only = true
noise_offset = 0.0
zero_terminal_snr = false
prior_loss_weight = 1.0
max_data_loader_n_workers = 8
bucket_reso_steps = 64
min_timestep = 0
max_timestep = 1000
v_pred_like_loss = 0.0
vae = "/workspace/Ktiseos-Nyx-Trainer/vae/sdxl_vae.safetensors"
network_module = "networks.lora"
network_dim = 32
network_alpha = 32
conv_dim = 32
conv_alpha = 32

There's also excess buttons that were introduced by me to do something specific that I was told was overbearing and overengineered, but it would mean that if those buttons were enabled PER tab: we could reload the toml upon refresh somehow.

The toml itself isn't mean to be a preset, so if there's a way to pre-detect toml information either by locating the config in the lora trainer details or based on what the dataset is?
